group/group.go

```diff
@@ -0,0 +1,74 @@
+package group
+
+import (
+	"crypto/elliptic"
+	"encoding"
+	"errors"
+	"io"
+
+	"github.com/cloudflare/circl/ecc/p384"
+)
+
+var (
+	// P256 is the group generated by P-256 elliptic curve.
+	P256 Group = wG{elliptic.P256()}
+	// P384 is the group generated by P-384 elliptic curve.
+	P384 Group = wG{p384.P384()}
+	// P521 is the group generated by P-521 elliptic curve.
+	P521 Group = wG{elliptic.P521()}
+)
+
+// Group si
+type Group interface {
+	NewElt() Element
+	NewScl() Scalar
+	Identity() Element
+	Generator() Element
+	Order() Scalar
+	Hashes([]byte) Hasher
```

Comment:
Mmm... not really sure this is the best way. 

It is difficult to see here:

```
	h2c := s.suite.Group.Hashes(dst)
	p := h2c.Hash(in)
```

what it is actually happening. Can this be refactored into one call to a `HashToGroup` func?

Reply:
Here, I followed the pattern of crypto.Hash, in which a hash function is first created, and then it can be used and reused in many invocations. This specific line, it uses h2c once.

Reply:
It's also unclear why this function is called `Hashes` -- what's the rationale?

Reply:
refactored.